### DISCUSSION TITLE
How to check if a key is already mapped?

### DISCUSSION THREAD

**Discussion by tahir-hassan** [helix-editor/helix#4796] [Q&A]
Hi, I don't want to "overbind" some useful built-in mapping, so is there a way to check if a key/combination is already mapped?

I was about to map `C-r` to `mip` and then `:reflow` for insert mode, but then fortunately tried `C-r` (which shows the registers).

Is there some way to list bound and unbound keys?

**Comment by Omnikar** [comment]
#997 could help with this, but you can also check [the documentation](https://docs.helix-editor.com/keymap.html).

**Comment by the-mikedavis** [comment]
The command picker (`<space>/`) shows commands and their keybinds. You can search within the command picker for a keybind

**Reply by tahir-hassan** [reply]
> The command picker (`<space>/`) shows commands and their keybinds. You can search within the command picker for a keybind

I  take it you mean the command pallette `<space>?` ?

Is there anything similar for Insert mode?

**Reply by the-mikedavis** [reply]
Ah whoops yep I mean the command palette `<space>?`. It doesn't list the insert mode bindings, the documentation would be the best resource for those.

**Reply by Omnikar** [reply]
> It doesn't list the insert mode bindings

Well… if you bind something to `command_palette` under `[keys.insert]` and use that, then it does. :P
